src/irr/asset/CCPUMeshPacker.cpp

```diff
@@ -0,0 +1,63 @@
+#include "irr/asset/CCPUMeshPacker.h"
+#include "irrlicht.h"
+
+namespace irr 
+{ 
+namespace asset 
+{
+
+std::optional<std::pair<ICPUMesh*, DrawElementsIndirectCommand_t>> CCPUMeshPacker::packMeshes(core::vector<ICPUMesh*>& meshes)
+{
+	core::vector<ICPUMeshBuffer*> meshBuffers;
+
+	for (auto* mesh : meshes)
+	{
+		for (uint32_t i = 0; i < mesh->getMeshBufferCount(); i++)
+		{
+			meshBuffers.push_back(mesh->getMeshBuffer(i));
+		}
+	}
+
+	if (meshBuffers.empty())
+	{
+		_IRR_DEBUG_BREAK_IF(true);
+		return {};
+	}
+
+	/*
+	Requirements for input mesh buffers:
+		- the same number and size of attributes
+		- all meshbufers have indexed triangle list (temporary)
+	*/
+
+	//validation
+	for (auto meshBuffer : meshBuffers)
+	{
+		auto* pipeline = meshBuffer->getPipeline();
+		auto* layout = pipeline->getLayout();
+
+		if (meshBuffer->getIndexBufferBinding()->buffer.get() == nullptr || 
```

Comment:
you need to handle nullptr index buffer, (its a std::iota starting from base index)

Reply:
hmm, this could be done by preallocating a 128kb range at the start of your storage buffer  that contains an std::iota of 0 to uint16_t_max!

Reply:
@Przemog1 you don't need to handle TRIANGLE_FAN, TRIANGLE_STRIP inputs right now, but you need to handle null index buffers